src/pages/Admin/AdminVenueRoomDetails.tsx

```diff
@@ -56,13 +56,13 @@ export const AdminVenueRoomDetails = ({
   const { user } = useUser();
   const history = useHistory();
 
-  const updateRoom = async (newState: boolean) => {
+  const updateRoom = async (newState: boolean, field: string) => {
```

Comment:
We don't really like to use multiple positional parameters like this. We should refactor this to take a 'props' object and use named keys here.

I'm also not sure that we should allow this to be a `string`, as that will make it too easy to accidentally overwrite things, and doesn't make the best use of our type system to guide us towards writing good code. As a minimal implementation here, we should change the type to `"isEnabled" | "isLabelled"`

Since we're changing it anyway, may as well rename this something better at the same time (eg. updateRoomBoolean

```suggestion
  export interface UpdateRoomBooleanProps {
    field: "isEnabled" | "isLabelled";
    newState: boolean;
  }
  
  const updateRoomBoolean = async ({ field: string, newState: boolean }) => {
```

Reply:
This function would also be best if it was moved into somewhere like `api/rooms.ts`

Reply:
Decided to call most of this OOS as this as it's not worth cleaning up since we plan to remove it as soon as admin v3 reaches feature parity.

fixed the typing in https://github.com/sparkletown/sparkle/pull/1632/commits/88062e4868edc765e7f9b6db71d8d76159c296a8